File: Finding middle element in a linked list/Finding middle element in a linked list.py
class Solution:
    #  Should return data of middle node. If linked list is empty, then  -1
    def findMid(self, head):
        current=head
        count=0
        while current is not None:
            count+=1
            current=current.next
        count=(count//2)+1
        temp=0
        while head!=None:
            temp+=1
            if count==temp:
                return head.data
            head=head.next

File: Finding middle element in a linked list/Finding middle element in a linked list.cpp
#include <iostream>
using namespace std;

class Solution{
    public:
    /* Should return data of middle node. If linked list is empty, then  -1*/

    int getMiddle(Node *head)
    {
        if(head->next==NULL) {
            return head->data;
        }
       Node*slow = head;
       Node* fast = head->next;
       while(fast!=NULL) {
           fast = fast->next;
           if(fast!=NULL) {
               fast = fast->next;
           }
           slow = slow->next;
       }
       return slow->data;
    }


};